Sources/BraveWallet/Panels/Signature Request/SignatureRequestView.swift

```diff
@@ -128,6 +190,17 @@ struct SignatureRequestView: View {
     .navigationBarTitleDisplayMode(.inline)
     .foregroundColor(Color(.braveLabel))
     .background(Color(.braveGroupedBackground).edgesIgnoringSafeArea(.all))
+    .introspectTextView { textView in
+      // A flash to show users message is overflowing the text view (related to issue https://github.com/brave/brave-ios/issues/6277)
+      if showOrignalMessage[requestIndex] == true {
```

Comment:
I believe this creates a tricky to repro UI bug where we show the warning when not needed, because `needPilcrowFormatted` is only updated when `showOriginalMessage[requestIndex]` is true.

STR:
1. Send request that requires warning message
2. Send request that requires no warning message
3. Open requests modal
4. Tap 'Show encoded message'
5. Tap 'Cancel' / 'Sign'

https://user-images.githubusercontent.com/5314553/199510393-df7756fa-a492-4c1a-b391-96e879413aa8.mov

I think resetting the 2 dictionaries when we approve/cancel (thus changing the indexes of the requests, which is used as the dictionary key) would resolve.